sdk/python/kfp/dsl/_component_bridge.py

```diff
@@ -18,31 +18,86 @@
 from ..components._components import _default_component_name, _resolve_command_line_and_paths
 from ..components._naming import _sanitize_python_function_name, generate_unique_name_conversion_table
 from .. import dsl
-
+from kfp.ir import pipeline_spec_pb2
 
 def _create_container_op_from_component_and_arguments(
     component_spec: ComponentSpec,
     arguments: Mapping[str, Any],
     component_ref: ComponentReference = None,
 ) -> 'dsl.ContainerOp':
+
+    pipeline_task_spec = pipeline_spec_pb2.PipelineTaskSpec()
+    pipeline_task_spec.task_info.name = component_spec.name
+
     # Check types of the reference arguments and serialize PipelineParams
     arguments = arguments.copy()
     for input_name, argument_value in arguments.items():
-        if isinstance(argument_value, dsl.PipelineParam):
+        if isinstance(argument_value, dsl.ContainerOp):
+            raise TypeError('ContainerOp object was passed to component as an input argument. Pass a single output instead.')
+        elif isinstance(argument_value, dsl.PipelineParam):
             input_type = component_spec._inputs_dict[input_name].type
             reference_type = argument_value.param_type
             dsl.types.verify_type_compatibility(reference_type, input_type, 'Incompatible argument passed to the input "{}" of component "{}": '.format(input_name, component_spec.name))
 
             arguments[input_name] = str(argument_value)
-        if isinstance(argument_value, dsl.ContainerOp):
-            raise TypeError('ContainerOp object was passed to component as an input argument. Pass a single output instead.')
 
+            # TODO: support more artifact types
+            if input_type.lower() == 'gcspath':
```

Comment:
Maybe
```python
if input_type.lower() in ['gcspath', 'gcsuri', uri]:
```

Reply:
synced offline. will start with gcspath for now.